### PR TITLE
[DOM-20310] Fix Model Version Publish Bug

### PR DESCRIPTION
**Pull request by abhijeet2096**

### CODE REVIEW

File: version.py [hunk 1]
```diff
@@ -0,0 +1 @@
+PYTHON_DOMINO_VERSION = '1.0.0'
```
Comment (vivekananda) [inline version.py:1]: PEP 8: no newline at end of file

File: domino/helpers.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+Helper function to check for version compatibility
+
+@:param version  Domino version to check version compatibility against
+
+@:exception throws exception if domino version isn't compatible with \
+            current python-domino version
+"""
+
+
+def is_version_compatible(version: str) -> None:
```
Comment (vivekananda) [inline domino/helpers.py:15]: The doc should be written as follows

def is_version_compatible(version: str) -> None:
    """
    Helper function to check for version compatibility

    @:param version  Domino version to check version compatibility against

    @:exception throws exception if domino version isn't compatible with \
                current python-domino version
    """

File: domino/domino.py [hunk 3]
```diff
@@ -41,12 +36,19 @@ def __init__(self, project, api_key=None, host=None):
                 constructor value or through DOMINO_USER_API_KEY environment \
                 variable.")
 
+        self._logger.info('Initializing Domino API with host ' + host)
+
+        owner_username = project.split("/")[0]
```
Comment (vivekananda) [inline domino/domino.py:41]: Minor change, why to do the same split twice instead take the split value into a variable

tokens = project.split("/")
owner_username = tokens[0]
project_name = tokens[1]

File: domino/constants.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+"""
+Minimum Domino version supported by this python-domino library
+"""
+
+MINIMUM_SUPPORTED_VERSION = '4.0.0'
```
Comment (vivekananda) [inline domino/constants.py:5]: I think `MINIMUM_SUPPORTED_DOMINO_VERSION` better conveys the meaning of the variable.

File: domino/domino.py [hunk 5]
```diff
@@ -380,17 +381,9 @@ def requires_at_least(self, at_least_version):
...
-
-        response = requests.get(url, auth=('', self._api_key))
-        regex = re.compile("/models/create\\?projectId=(.{24,24})")
-        matches = regex.findall(response.text)
-        if len(matches) > 0:
-            return matches[0]
-        else:
-            return None
+        url = self._routes.find_project_by_owner_name_and_project_name()
+        project_id = self._get(url)['id']
```
Comment (vivekananda) [inline domino/domino.py:385]: What if the response doesn't have `id` in it?

**Comment by abhijeet2096** [comment]
Closing this In favour of this https://github.com/dominodatalab/python-domino/pull/53

**Pull request closed without merging**